compiler/test/dotc/tests.scala

```diff
@@ -167,6 +167,7 @@ class tests extends CompilerTest {
   @Test def rewrites = compileFile(posScala2Dir, "rewrites", "-rewrite" :: scala2mode)
 
   @Test def pos_t8146a = compileFile(posSpecialDir, "t8146a")(allowDeepSubtypes)
+  @Test def pos_jon = compileFile(posSpecialDir, "jon")(allowDeepSubtypes)
```

Comment:
I did not look too deeply. But SeqFactory has a funky recursive type

    SeqFactory[CC[X] <: Seq[X] with GenericTraversableTemplate[X, CC]] extends GenSeqFactory[CC] with TraversableFactory[CC]

so it's not so implausible.